pageserver/src/tenant/timeline.rs

```diff
@@ -4866,7 +4932,7 @@ impl Timeline {
                 // because we have not implemented L0 => L0 compaction.
                 duplicated_layers.insert(l.layer_desc().key());
             } else if LayerMap::is_l0(&l.layer_desc().key_range) {
-                bail!("compaction generates a L0 layer file as output, which will cause infinite compaction.");
+                return Err(CompactionError::GeneratedL0);
```

Comment:
This can be manual `CompactionError::Other()`. Nobody cares about the difference between `CompactionError::GeneratedL0` and `CompactionError::Other` => just needless `match` bureaucracy at the places that inspect `CompationError`s.

Reply:
I see you already changed it in another PR :D I removed it.